include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -482,41 +492,487 @@ namespace internal
       , active_fe_index_function(active_fe_index_function)
     {}
 
+    /**
+     * Return if cell has child.
+     */
     bool
     has_children() const
     {
       return has_children_function();
     }
 
+    /**
+     * Get DoF indices.
+     */
     void
     get_dof_indices(std::vector<types::global_dof_index> &dof_indices) const
     {
       get_dof_indices_function(dof_indices);
     }
-
+    /**
+     * Get active FE index.
+     */
     unsigned int
     active_fe_index() const
     {
       return active_fe_index_function();
     }
 
-
   private:
+    /**
+     * Lambda function returning if cell has child.
+     */
     const std::function<bool()> has_children_function;
+
+    /**
+     * Lambda function returning DoF indices.
+     */
     const std::function<void(std::vector<types::global_dof_index> &)>
-                                        get_dof_indices_function;
+      get_dof_indices_function;
+
+    /**
+     * Lambda function returning active FE index.
+     */
     const std::function<unsigned int()> active_fe_index_function;
   };
 
 
 
+  /**
+   * Base class for a view on fine DoFHandler.
+   *
+   * Implementations include:
+   *  - IdentityFineDoFHandlerView: all cells on the fine mesh are either
+   *    locally owned or ghosted; useful for p-multigrid without repartitioning;
+   *  - FistChildPolicyFineDoFHandlerView: parent cells are owned by the first
+   *    child cell; useful for local smoothing with fast setup;
+   *  - PermutationFineDoFHandlerView: fine mesh has the same set as the coarse
+   *    mesh but are partitioned differently; useful for p-multigrid with
+   *    repartitioning;
+   *  - GlobalCoarseningFineDoFHandlerView: cells on the coarse mesh are either
+   *    refined or not; useful for global coarsening.
+   */
+  template <int dim>
+  class FineDoFHandlerViewBase
+  {
+  public:
+    virtual ~FineDoFHandlerViewBase() = default;
+
+    /**
+     * Return cell on fine DoFHandler.
+     */
+    virtual FineDoFHandlerViewCell
+    get_cell(const typename DoFHandler<dim>::cell_iterator &cell) const = 0;
```

Comment:
I struggle with understanding this name. Wouldn't it be better to use
```suggestion
    get_view(const typename DoFHandler<dim>::cell_iterator &cell) const = 0;
```